packages/product-json-to-csv/src/main.js

```diff
@@ -0,0 +1,292 @@
+/* @flow */
+import type {
+  ApiConfigOptions,
+  Category,
+  LoggerOptions,
+  ParserConfigOptions,
+  ProductProjection,
+  ProductType,
+  ResolvedProdProj,
+  State,
+  TaxCategory,
+  TypeReference,
+} from 'types/product'
+import type { Client, ClientRequest, SuccessResult } from 'types/sdk'
+
+import { createClient } from '@commercetools/sdk-client'
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import { createAuthMiddlewareForClientCredentialsFlow } from '@commercetools/sdk-middleware-auth'
+import { createUserAgentMiddleware } from '@commercetools/sdk-middleware-user-agent'
+import highland from 'highland'
+import Promise from 'bluebird'
+import { memoize } from 'lodash'
+import ProductMapping from './map-product-data'
+import { writeToSingleCsvFile, writeToZipFile } from './writer'
+import pkg from '../package.json'
+
+export default class ProductJsonToCsv {
+  // Set flowtype annotations
+  accessToken: string
+  apiConfig: ApiConfigOptions
+  categoriesCache: Object
+  client: Client
+  parserConfig: ParserConfigOptions
+  logger: LoggerOptions
+  fetchReferences: Function
+  _resolveReferences: Function
+  _productMapping: Object
+
+  constructor(
+    apiConfig: ApiConfigOptions,
+    parserConfig: ParserConfigOptions,
+    logger: LoggerOptions,
+    accessToken: string
+  ) {
+    this.apiConfig = apiConfig
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: pkg.name,
+          libraryVersion: pkg.version,
+        }),
+        createHttpMiddleware({ host: this.apiConfig.apiUrl }),
+      ],
+    })
+
+    const defaultConfig = {
+      categoryBy: 'name', // key, externalId or namedPath supported
+      categoryOrderHintBy: 'name', // key, externalId or name supported
+      delimiter: ',',
+      fillAllRows: false,
+      language: 'en',
+      multiValueDelimiter: ';',
+      productSeparator: '\n',
+    }
+
+    this.parserConfig = { ...defaultConfig, ...parserConfig }
+    this.logger = {
```

Comment:
The logger is passed in from [L74](https://github.com/commercetools/nodejs/pull/326/files/3ecd2c2f940eb11a575140e0e2c267aa46757765#diff-e0a3628cb03c02182cfe6b4c1295da09R74). It is done this way so that in the event that a custom logger is passed in without any of the functions, it will not throw while trying to log